hal/lib/athena/DIO.cpp

```diff
@@ -67,10 +67,9 @@ HAL_DigitalHandle HAL_InitializeDIOPort(HAL_PortHandle port_handle,
           outputEnable.Headers | bitToSet;  // set the bit for write
     }
   } else {
-    uint32_t bitToSet = 1 << remapMXPChannel(port->pin);
+    uint32_t bitToSet = 1u << remapMXPChannel(port->pin);
 
-    // Disable special functions on this pin
-    int16_t specialFunctions =
+    unsigned short specialFunctions =
```

Comment:
Sigh, this will probably cause cpplint to complain, but you might get a compiler warning using uint16_t?